monai/transforms/intensity/array.py

```diff
@@ -1137,3 +1131,164 @@ def __call__(self, img: np.ndarray) -> np.ndarray:
         return np.asarray(
             np.interp(img, reference_control_points_scaled, floating_control_points_scaled), dtype=img.dtype
         )
+
+
+class RandGibbsNoise(RandomizableTransform):
+    """
+    Naturalistic image augmentation via Gibbs artifacts. The transform 
+    randomly applies Gibbs noise to 2D/3D MRI images. Gibbs artifacts
+    are one of the common type of type artifacts appearing in MRI scans.
+
+    The transform is applied to all the channels in the data.
+    
+    For general information on Gibbs artifacts, please refer to:
+    https://pubs.rsna.org/doi/full/10.1148/rg.313105115
+    https://pubs.rsna.org/doi/full/10.1148/radiographics.22.4.g02jl14949
+
+
+    Args:
+        prob (float): probability of applying the transform.
+        alpha (float, List(float)): Parametrizes the intensity of the Gibbs noise filter applied. Takes
+            values in the interval [0,1] with alpha = 0 acting as the identity mapping. 
+            If a length-2 list is given as [a,b] then the value of alpha will be 
+            sampled uniformly from the interval [a,b]. 0 <= a <= b <= 1.
+        dim : spatial dimensions of the image (with any leading dimensions). 
+            Set dim = 2 for 2D images, and dim = 3 for 3D images.
+    """
+
+    def __init__(self, prob: float = 0.1, alpha: Union[float, List[float]] = 0.5, dim: int = 3) -> None:
+
+        if type(alpha) == list:
+            if alpha[1] > 1 or alpha[0] < 0:
+                raise AssertionError("alpha must take values in the interval [0,1]")
+            if alpha[0] > alpha[1]:
+                raise AssertionError("When alpha = [a,b] we need a < b.")
+            if len(alpha) != 2:
+                raise AssertionError("If alpha is a list, then its length must be 2.")
+        else:
+            if alpha > 1 or alpha < 0:
+                raise AssertionError("alpha must take values in the interval [0,1].")
+
+        if dim not in [2, 3]:
+            raise AssertionError(("Set dim = 2 or dim = 3 depending on spatial" " dimensions of image."))
+
+        self.alpha = alpha
+        self.dim = dim
+        self.sampled_alpha = None  # stores last alpha sampled by randomize()
+        RandomizableTransform.__init__(self, prob=prob)
+
+    def __call__(self, img: torch.tensor) -> torch.tensor:
+
+        # randomize application and possibly alpha
+        self.sampled_alpha = self.randomize()
```

Comment:
I'm not sure that we'd want to store the most recent alpha as this might lead to multiprocessing problems. Is that correct @Nic-Ma ?

Reply:
My initial thinking to store the alpha was in case the user wanted to know which alpha was sampled and applied to the data.

Reply:
I think you're right. Plus, `randomize` doesn't allow us to return anything so we don't have much choice anyway!